exec/java-exec/src/main/java/org/apache/calcite/jdbc/DynamicRootSchema.java

```diff
@@ -117,6 +119,14 @@ public void loadSchemaFactory(String schemaName, boolean caseSensitive) {
       }
     } catch(ExecutionSetupException | IOException ex) {
       logger.warn("Failed to load schema for \"" + schemaName + "\"!", ex);
+      // We can't proceed further without a schema, throw a runtime exception.
+      UserException.Builder exceptBuilder =
+          UserException
+              .resourceError(ex)
```

Comment:
should be `executionError` instead of `resourceError`.

Reply:
The reason I chose `resourceError` is that schema as a resource is unavailable and it requires a user to externally resolve the issue. Could you elaborate on why this should be `executionError`?

Reply:
Had an offline discussion on this that IOException from different plugins is overloading different IOException category. ResourceError being most common category we are going with it for now.